alignas (max_align_t) compilation error for x86_64 with glibc-2.21

Hi,

Seen below error while compiling for x86_64 using glibc 2.21 same was working good with glibc 2.27

OpenWrt-libtool: compile:  ccache_cc -DHAVE_CONFIG_H -I. -I./lib -DIN_LIBCRYPT -Istaging_dir/target-x86_64_glibc-2.21/usr/include -Istaging_dir/target-x86_64_glibc-2.21/include -Istaging_dir/toolchain-x86_64_gcc-4.8-linaro_glibc-2.21/usr/include -Istaging_dir/toolchain-x86_64_gcc-4.8-linaro_glibc-2.21/include -Wall -Wextra -Wbad-function-cast -Wcast-align -Wcast-qual -Wconversion -Wformat=2 -Wlogical-op -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wold-style-definition -Wpointer-arith -Wshadow -Wstrict-overflow=2 -Wstrict-prototypes -Wundef -Wvla -Wwrite-strings -Werror -fno-plt -Os -pipe -g3 -fno-caller-saves -fhonour-copts -Wno-error=unused-but-set-variable -Wall -pipe -Wformat -Werror=format-security -Wl,-z,now -Wl,-z,relro -Wno-error=long-long -Wno-error=pedantic -std=c99 -MT lib/libxcrypt_la-crypt.lo -MD -MP -MF lib/.deps/libxcrypt_la-crypt.Tpo -c lib/crypt.c  -fPIC -DPIC -o lib/.libs/libxcrypt_la-crypt.o
In file included from lib/crypt-port.h:27:0,
                 from lib/crypt.c:20:
_lib/crypt.c:35:17: error: expected expression before 'max_align_t'
   char alignas (max_align_t) alg_specific[ALG_SPECIFIC_SIZE];
                 ^
./config.h:214:47: note: in definition of macro 'alignas'
 #define alignas(n) __attribute__((__aligned__(n)))
                                               ^
./config.h:214:49: error: expected identifier or '(' before ')' token
 #define alignas(n) __attribute__((__aligned__(n)))
                                                 ^
lib/crypt.c:35:8: note: in expansion of macro 'alignas'
   char alignas (max_align_t) alg_specific[ALG_SPECIFIC_SIZE];
        ^
lib/crypt.c:36:1: error: no semicolon at end of struct or union [-Werror]
 };_
 ^
lib/crypt.c: In function 'do_crypt':
lib/crypt.c:142:17: error: 'struct crypt_internal' has no member named 'alg_specific'
             cint->alg_specific, sizeof cint->alg_specific);
                 ^
lib/crypt.c:142:44: error: 'struct crypt_internal' has no member named 'alg_specific'
             cint->alg_specific, sizeof cint->alg_specific);
                                            ^
cc1: all warnings being treated as errors

Found one of the usage indicating _alignas(expression)_ Equivalent to _alignas(alignof(type))_
So making the corresponding change as mentioned below worked

`
--- a/lib/crypt.c
+++ b/lib/crypt.c
@@ -32,7 +32,7 @@
    struct, since the first field must always have offset 0.  */
 struct crypt_internal
 {
\- char alignas (max_align_t) alg_specific[ALG_SPECIFIC_SIZE];
\+ char alignas(alignof(max_align_t)) alg_specific[ALG_SPECIFIC_SIZE];
 };
 
 static_assert(sizeof (struct crypt_internal) + alignof (struct crypt_internal)
`
Any impact with above change.

Thanks,
Punith Kumar

--- a/NEWS
+++ b/NEWS
@@ -4,6 +4,8 @@ Please send bug reports, questions and suggestions to
 <https://github.com/besser82/libxcrypt/issues>.
 
 Version 4.4.17
+* Fix compilation error in 'alignas (type)' with older versions
+  of glibc and/or gcc (issue #107).
 
 Version 4.4.16
 * Add support for the e2k architecture.

--- a/lib/crypt.c
+++ b/lib/crypt.c
@@ -32,7 +32,7 @@
    struct, since the first field must always have offset 0.  */
 struct crypt_internal
 {
-  char alignas (max_align_t) alg_specific[ALG_SPECIFIC_SIZE];
+  char alignas (alignof (max_align_t)) alg_specific[ALG_SPECIFIC_SIZE];
 };
 
 static_assert(sizeof (struct crypt_internal) + alignof (struct crypt_internal)

